File: backend/Backend/src/main/java/ru/nikidzawa/backend/store/client/dto/IndividualDtoShort.java
package ru.nikidzawa.backend.store.client.dto;

import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class IndividualDtoShort {
    Long id;
    String nickname;
    String name;
    String surname;
    LocalDate birthdate;
    String aboutMe;
    LocalDateTime lastOnline;
    String avatarHref;
}

File: frontend/react-app/src/api/internal/dto/IndividualDtoShort.js
export default class IndividualDtoShort {
    constructor(id, nickname, name, surname, birthdate, aboutMe, avatarHref, lastLogoutDate) {
        this.id = id;
        this.nickname = nickname;
        this.name = name;
        this.surname = surname || '';
        this.birthdate = birthdate;
        this.aboutMe = aboutMe;
        this.avatarHref = avatarHref;
        this.lastLogoutDate = lastLogoutDate;
    }

    static fromJSON(data) {
        return new IndividualDtoShort(
            data.id,
            data.nickname,
            data.name,
            data.surname || '',
            data.birthdate,
            data.aboutMe,
            data.avatarHref,
            new Date(data.lastLogoutDate)
        );
    }
}